lib/mirage_key.ml

```diff
@@ -44,12 +44,38 @@ module Arg = struct
   let builtin d mn m = of_module (from_run d) mn m
 
   let ipv4 = builtin "ipv4" "Ipaddr.V4" (module Ipaddr.V4)
-  let ipv4_prefix =
-    builtin "ipv4_prefix" "Ipaddr.V4.Prefix" (module Ipaddr.V4.Prefix)
   let ipv6 = builtin "ipv6" "Ipaddr.V6" (module Ipaddr.V6)
   let ipv6_prefix =
     builtin "ipv6_prefix" "Ipaddr.V6.Prefix" (module Ipaddr.V6.Prefix)
 
+  let ipv4_network =
+    let serialize fmt (ip, prefix) =
+      Format.fprintf fmt "((Ipaddr.V4.of_string_exn \"%a\"), (Ipaddr.V4.Prefix.of_string_exn \"%a/%d\"))"
+      Ipaddr.V4.pp_hum ip
+      Ipaddr.V4.pp_hum ip
+      (Ipaddr.V4.Prefix.bits prefix)
+    in
+    let print fmt (ip, prefix) = 
+      Format.fprintf fmt "%a/%d" Ipaddr.V4.pp_hum ip (Ipaddr.V4.Prefix.bits prefix)
+    in
+    let parse str =
+      let (>>=) x f =
+        match x with
+        | None -> `Error (str ^ " is not a valid IPv4 address and netmask")
+        | Some g -> f g
+      in
+      Ipaddr.V4.Prefix.of_string str >>= fun network ->
+      (* recover the IP *)
+      Astring.String.cut ~sep:"/" str >>= fun (ip, _) ->
+      Ipaddr.V4.of_string ip >>= fun ip ->
+      `Ok (ip, network)
```

Comment:
`Ipaddr.V4.Prefix.of_address_string` does exactly this without the ad hoc string parsing.